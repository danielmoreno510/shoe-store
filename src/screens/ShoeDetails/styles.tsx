import {StyleSheet} from 'react-native';
import {widthPercentageToDP} from '../../utils/responsiveUtils';

export default StyleSheet.create({
  container: {
    padding: widthPercentageToDP('5%'),
    flex: 1,
    backgroundColor: 'white',
  },
});
