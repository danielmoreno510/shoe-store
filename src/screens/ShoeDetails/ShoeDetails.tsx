import React from 'react';
import {View} from 'react-native';

import Carousel from '../../components/common/Shoe/Carousel';
import ShoeInfo from '../../components/common/Shoe/ShoeInfo';
import styles from './styles';

const ShoesDetails: React.FC<ShoesDetailsProps> = ({currentShoe, cart, setAddToCart}) => (
  <View style={styles.container}>
    <Carousel shoe={currentShoe} width={300} />
    <ShoeInfo shoe={currentShoe} cart={cart} addToCart={setAddToCart} twoSection />
  </View>
);

export default ShoesDetails;
