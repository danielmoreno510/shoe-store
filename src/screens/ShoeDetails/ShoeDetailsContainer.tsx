import ShoeDetails from './ShoeDetails';
import {StoreService} from '../../redux/StoreService';
import {putAddToCart} from '../../redux/actions';

export const mapStateToProps = state => ({
  currentShoe: state.shoes.currentShoe,
  cart: state.shoes.cart,
});

export const mapDispatchToProps = dispatch => ({
  setAddToCart: shoe => {
    dispatch(putAddToCart(shoe));
  },
});

const ShoeDetailsContainer = StoreService.connect(mapStateToProps, mapDispatchToProps)(ShoeDetails);

export default ShoeDetailsContainer;
