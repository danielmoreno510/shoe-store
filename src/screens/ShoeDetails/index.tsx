import ShoeDetailsContainer from './ShoeDetailsContainer';

export default ShoeDetailsContainer;
