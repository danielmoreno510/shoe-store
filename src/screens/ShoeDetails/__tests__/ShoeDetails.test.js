import React from 'react';
import {shallow} from 'enzyme';

import ShoeDetails from '../ShoeDetails';

describe('ShoeDetails component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      currentShoe: {},
      cart: [],
      setAddToCart: jest.fn(),
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<ShoeDetails {...testProps} />);

      it('THEN should display a regular ShoeDetails', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
