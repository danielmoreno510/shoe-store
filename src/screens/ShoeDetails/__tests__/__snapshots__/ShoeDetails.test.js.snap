// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoeDetails component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular ShoeDetails 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 37.5,
    }
  }
>
  <CustomCarousel
    shoe={Object {}}
    width={300}
  />
  <ShoeInfo
    addToCart={[MockFunction]}
    cart={Array []}
    shoe={Object {}}
    twoSection={true}
  />
</View>
`;
